### PR TITLE
fix: python bindings and examples

### PR DESCRIPTION
**Pull request by qrayven**
# Description of change

 - fix  `prepare_output` which was returning the `Output` class causing:
 - fix `Client::build_*` to return output of specific type
 -  minor fixes to examples
```
Traceback (most recent call last):
  File "/code/rayven/iota/iota-sdk/bindings/python/examples/wallet/12-prepare_output.py", line 30, in <module>
    transaction = account.send_outputs([output])
  File "/code/rayven/iota/iota-sdk/bindings/python/.venv/lib/python3.10/site-packages/iota_sdk/wallet/account.py", line 513, in send_outputs
    return Transaction.from_dict(self._call_account_method(
  File "/code/rayven/iota/iota-sdk/bindings/python/.venv/lib/python3.10/site-packages/iota_sdk/wallet/common.py", line 53, in wrapper
    response = call_wallet_method(args[0].handle, message)
ValueError: cannot deserialize basic output: missing field `amount` at line 1 column 122
```

Please write a summary of your changes and why you made them. 

## Links to any relevant issues
part of #388 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/python/iota_sdk/types/output.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 from dataclasses import dataclass, field
 from enum import IntEnum
 from typing import Dict, Optional, List
+
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/output.py]:
```suggestion
```

File: bindings/python/iota_sdk/wallet/account.py [hunk 2]
```diff
@@ -343,12 +343,12 @@ def prepare_output(self, params: OutputParams,
            When the assets contain an nft_id, the data from the existing nft output will be used, just with the address
            unlock conditions replaced
         """
-        return from_dict(Output, self._call_account_method(
+        return output_from_dict(self._call_account_method(
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/wallet/account.py:346]: Changelog that this fixes prepare_output()

Comment (qrayven) [reply bindings/python/iota_sdk/wallet/account.py:346]: done

File: bindings/python/iota_sdk/wallet/account.py [hunk 3]
```diff
@@ -335,7 +335,7 @@ def get_balance(self) -> Balance:
         ))
 
     def prepare_output(self, params: OutputParams,
-                       transaction_options: Optional[TransactionOptions] = None):
+                       transaction_options: Optional[TransactionOptions] = None) -> TreasuryOutput | AliasOutput | BasicOutput | FoundryOutput | NftOutput | Output:
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/wallet/account.py]: prepare_output() can only return Basic or NFT outputs, so should be this?
```suggestion
                       transaction_options: Optional[TransactionOptions] = None) -> BasicOutput | NftOutput:
```

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/python/examples/how_tos/outputs/features.py [hunk 4]
```diff
@@ -73,4 +73,4 @@
 )
 outputs.append(nft_output)
 
-print(json.dumps(outputs, indent=2))
+print(json.dumps([asdict(o) for o in outputs], indent=2))
```
Comment (Thoralf-M) [inline bindings/python/examples/how_tos/outputs/features.py:77]: Doesn't this need the `from dataclasses import asdict` import?

Comment (qrayven) [reply bindings/python/examples/how_tos/outputs/features.py:77]: it's weird It was working without it :D.  But I've added it just to make sure

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**